### PR TITLE
LUM-13241 - test on basic projects

### PR DESCRIPTION
**Pull request by Blackbaud-BethRichardson**
This ensures that the basic and REST projects build and run successfully as built by gradle-templates. This means that bootRun and clean check run for both types of projects and that builds will pass in jenkins without much intervention (you must supply a port besides the default of -1)

@Blackbaud-MikeLueders @blackbaud/team-cerebro

### CODE REVIEW

File: src/main/resources/templates/springboot/application-class.java.tmpl [hunk 1]
```diff
@@ -9,8 +9,7 @@ import org.springframework.boot.autoconfigure.security.SecurityAutoConfiguration
 import org.springframework.boot.orm.jpa.EntityScan;
 import org.springframework.context.annotation.ComponentScan;
 
-@SpringBootApplication
-@EntityScan({"${servicePackage}", "com.blackbaud.boot.converters"})
+@SpringBootApplication${entityScan}
```
Comment (Blackbaud-MikeLueders) [inline src/main/resources/templates/springboot/application-class.java.tmpl]: it's a bit odd having ${entityScan} at the end of line and relying on the \n in the entityScan text... consider just putting ${entityScan} on the line above and having an extra space if you don't have the entity scan... the other option would be to generate the basic template and then using FileUtils.appendAfterLine to append the EntityScan line after SpringBootApplication

File: src/main/resources/templates/springboot/rest/build.gradle.tmpl [hunk 2]
```diff
@@ -32,12 +32,18 @@ dependencies {
...
     clientCompile "com.blackbaud:common-spring-boot-rest:\${springBootVersion}-2.+"
 
+    mainTestCompile "com.blackbaud:common-spring-boot-rest-test:\${springBootVersion}-2.+"
     mainTestCompile "com.blackbaud:common-test:1.+"
     mainTestCompile "org.codehaus.groovy:groovy-all:2.4.3"
     mainTestCompile ("org.spockframework:spock-spring:1.0-groovy-2.4") {
         exclude group: "org.codehaus.groovy"
     }
 
+    testCompile ('org.spockframework:spock-spring:1.0-groovy-2.4') {
```
Comment (Blackbaud-MikeLueders) [inline src/main/resources/templates/springboot/rest/build.gradle.tmpl]: this is not necessary - testCompile inherits all dependencies from mainTestCompile

File: src/main/resources/templates/springboot/rest/build.gradle.tmpl [hunk 3]
```diff
@@ -32,12 +32,18 @@ dependencies {
...
+        exclude group: 'org.codehaus.groovy'
+    }
+    testCompile "org.springframework.boot:spring-boot-starter-test"
+
     componentTestCompile "org.springframework.boot:spring-boot-starter-test"
```
Comment (Blackbaud-MikeLueders) [inline src/main/resources/templates/springboot/rest/build.gradle.tmpl]: just change these two to be mainTestCompile and they'll be pulled in for both unit and component tests

**Review by Blackbaud-MikeLueders — APPROVED** [review]

**Comment by Blackbaud-BethRichardson** [comment]
@blackbaud/team-cerebro This should be ready to go.

**Pull request merged**